### ISSUE TITLE
Wiki: Comparison between Flutures and RxJS Observables

### ISSUE THREAD

**Issue by Lube** [Avaq/Fluture#7]
Been into RxJS the last few days and Observables seem so similar to flutures, any idea if there is a possibility to explain the difference between them?

**Comment by stoeffel** [comment]
The main thing to remember IMO is that observables are values over time where a Future is a value in the future.
Meaning by subscribing to an observable you will get `n` values at some point and by forking a Future you will get **one** value at some point.

**Comment by Avaq** [comment]
> by forking a Future you will get **one** value at some point

Though this is what Futures are meant to be used for, it's not always true. The following code:

```js
Future((rej, res) => setInterval(res, 333, 'world'))
.map(x => `Hello ${x}!`)
.fork(console.error, console.log)
```

Produces a "stream" of three `Hello world!`s per second. This behavior is unspecified though, and may change in the future. Yet currently, if you'd rename `fork` to `subscribe`, you wouldn't see the difference anymore. But there are subtle differences:

Currently a difference is that futures offer no means to "unsubscribe". However, this [may change when cancellation is introduced](https://github.com/Avaq/Fluture/tree/cancellation#cancellation-and-resource-disposal).

Futures also have no notion of "completion". The continuation is expected to run only once, so it may assume completion after it calls the continuation. This is similar to Continuation Passing Style (callbacks).

Furthermore, futures are lazy and stateless. Calling `fork` creates the source and "pulls" the value through the computation. To illustrate a consequence of this difference, let's modify the example from above:

```js
const source =
  Future((rej, res) => setInterval(res, 333, 'world'))
  .map(x => `Hello ${x}!`)

source.fork(console.error, console.log)
source.fork(console.error, console.log)
```

This would log six `Hello world!`s per second to the console. The key difference is that this happens *not because there are two subscribers to one source*, but because *there are two sources each with one continuation*. This is the most notable difference I think, and I've illustrated it in [this notebook](https://tonicdev.com/avaq/observables-vs-futures).

These differences are each very subtle, and I think the main difference lies in how we think about Futures versus Observables: Observables are to be used as "streams" carrying multiple values, whereas Futures are to be used as an abstraction over Continuation Passing Style, carrying a single value from one asynchronous computation to the next. The interesting thing to me is that you could probably point to an observable and tell everyone "I'm going to treat this like a Future", and for the majority of use-cases you'd be fine.

**Comment by Lube** [comment]
@Avaq Thanks that notebook was very helpful to illustrate your point, and it also look so sick!

**Comment by Avaq** [comment]
@Lube @stoeffel You might be interested in #17, and I would like to gather some feedback. :)

**Comment by Avaq** [comment]
New issue: https://github.com/Avaq/Fluture/issues/18

**Comment by Avaq** [comment]
Starting at [Version 2.0.0](https://github.com/Avaq/Fluture/releases/tag/2.0.0):

> by forking a Future you will get **one** value at some point

Is a true statement.
